### PR TITLE
Sync accounts

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift [hunk 1]
```diff
@@ -313,15 +301,15 @@ import PromiseKit
         // Add missing linkedBrokers
         let localUserIds = self.linkedBrokers.flatMap { $0.linkedLogin.userId }
         let remoteLinkedBrokersToAdd = remoteLinkedBrokers.filter { !localUserIds.contains($0.userId) }
-        
+
         remoteLinkedBrokersToAdd.forEach { remoteBrokerData in
-            injectBroker(
+            self.saveLinkedBrokerToKeychain(
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: These were confusing. I thought initially these were async calls and was about to promisify everything but appears that it's actually sync.

File: TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift [hunk 2]
```diff
@@ -331,10 +319,38 @@ import PromiseKit
...
+                syncAccounts(localLinkedBroker: localLinkedBroker, remoteLinkedBroker: remoteLinkedBroker)
+            }
+        }
+
         onFinished()
     }
 
+    private func syncAccounts(localLinkedBroker: TradeItLinkedBroker, remoteLinkedBroker: LinkedBrokerData) {
+        // Add missing accounts
+        let localAccountNumbers = localLinkedBroker.accounts.map { $0.accountNumber }
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: isn't `flatMap` ?

File: TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift [hunk 3]
```diff
@@ -331,10 +319,38 @@ import PromiseKit
...
             self.removeBroker(linkedBroker: linkedBrokerToRemove)
         }
-        
+
+        // Sync accounts
+        self.linkedBrokers.forEach { localLinkedBroker in
+            remoteLinkedBrokers.first { remoteLinkedBroker in
+                localLinkedBroker.linkedLogin.userId == remoteLinkedBroker.userId
+            }.flatMap { remoteLinkedBroker in
+                syncAccounts(localLinkedBroker: localLinkedBroker, remoteLinkedBroker: remoteLinkedBroker)
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: We want also to update userToken (see comments in here : https://github.com/tradingticket/AndroidSDK/pull/63)

Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: @guillaumedebavelaere are you sure? That means we'll need to read from the keychain every time. Yahoo plans to run this method every 30 seconds so it'll be very heavy.

Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift]: @guillaumedebavelaere I'm going to leave this as is for now. Yahoo doesn't need to check the `userToken`

**Pull request merged**